include/dlaf/factorization/cholesky/impl.h

```diff
@@ -68,6 +67,33 @@ void gemmTrailingMatrixTile(Executor&& trailing_matrix_executor,
                 std::move(matrix_tile));
 }
 
+template <class Executor, Device device, class T>
+void trsmPanelTileU(Executor&& executor_hp, hpx::shared_future<matrix::Tile<const T, device>> kk_tile,
+                    hpx::future<matrix::Tile<T, device>> matrix_tile) {
+  hpx::dataflow(executor_hp, matrix::unwrapExtendTiles(tile::trsm_o), blas::Side::Left,
+                blas::Uplo::Upper, blas::Op::ConjTrans, blas::Diag::NonUnit, T(1.0), std::move(kk_tile),
+                std::move(matrix_tile));
+}
+
+template <class Executor, Device device, class T>
+void herkTrailingDiagTileU(Executor&& trailing_matrix_executor,
+                           hpx::shared_future<matrix::Tile<const T, device>> panel_tile,
+                           hpx::future<matrix::Tile<T, device>> matrix_tile) {
+  hpx::dataflow(trailing_matrix_executor, matrix::unwrapExtendTiles(tile::herk_o), blas::Uplo::Upper,
+                blas::Op::ConjTrans, BaseType<T>(-1.0), panel_tile, BaseType<T>(1.0),
+                std::move(matrix_tile));
+}
+
+template <class Executor, Device device, class T>
+void gemmTrailingMatrixTileU(Executor&& trailing_matrix_executor,
+                             hpx::shared_future<matrix::Tile<const T, device>> panel_tile,
+                             hpx::shared_future<matrix::Tile<const T, device>> col_panel,
+                             hpx::future<matrix::Tile<T, device>> matrix_tile) {
+  hpx::dataflow(trailing_matrix_executor, matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::ConjTrans,
+                blas::Op::NoTrans, T(-1.0), std::move(panel_tile), std::move(col_panel), T(1.0),
+                std::move(matrix_tile));
+}
+
```

Comment:
As different algorithm in the same namespace might might need helper functions with the same name, which are not fully equivalent, I suggest to embed them in the namespaces
`dlaf::factorization::internal::cholesky_L` and `dlaf::factorization::internal::cholesky_U`
similarly to what has been done in #392 for the triangular solver:
https://github.com/eth-cscs/DLA-Future/blob/37e4c461a0f83011f384b2d3892a79e287add2f4/include/dlaf/solver/triangular/impl.h#L36-L55
https://github.com/eth-cscs/DLA-Future/blob/37e4c461a0f83011f384b2d3892a79e287add2f4/include/dlaf/solver/triangular/impl.h#L60